MockDuck/Sources/MockRequest.swift

```diff
@@ -13,6 +13,26 @@ import Foundation
 final class MockRequest {
     var request: URLRequest
 
+    private(set) lazy var normalizedRequest: URLRequest = {
+        return MockDuck.delegate?.normalizedRequest(for: request) ?? request
+    }()
+
+    var serializedHashValue: String {
+        let normalizedRequest = self.normalizedRequest
+
+        var hashData = Data()
+
+        if let urlData = normalizedRequest.url?.absoluteString.data(using: .utf8) {
+            hashData.append(urlData)
+        }
+
+        if let body = normalizedRequest.httpBody {
+            hashData.append(body)
+        }
+
```

Comment:
What do you think about adding

```Swift
if let method = normalizedRequest.httpMethod?.data(using: .utf8) {
            hashData.append(method)
        }
``` 
here to distinguish request with same url but different methods?

Reply:
Another good idea. Would break existing recordings so we would want to do it in a major version change. Maybe we work on a list of things that are not currently hashed that should be and do them all at once?

Reply:
I can open a new Pr for this and we can discuss what things should be added to hash there